src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Server/Program.cs

```diff
@@ -20,10 +20,10 @@ public static void Main(string[] args)
 
 			app.UseAuthorization();
 
-$if$($UseWebAssembly$ == True)
+#if USE_WASM
```

Comment:
This change does not appear to be in line with the original issue. The problem is that there was an invalid template condition. where `<!--#if` should have just been `#if`. This PR seems to be introducing a completely unnecessary MSBuild compiler constant which has nothing to do with the issue.
![image](https://user-images.githubusercontent.com/3860573/216770481-5b3f1473-9f76-4556-8d20-7991e03c484c.png)
This would make no sense to have here, and since the unoapp template actually removes the Server project if WASM isn't included I would assume here that the correct action is actually to remove the condition entirely. @jeromelaban what do you think?

Reply:
I agree the new MSBuild property is unnecessary, it just seemed to be quite easier to do to fix the issue. I can try with another approach if @jeromelaban thinks it's worth it.

Reply:
The origin of the condition is that it's possible to create the server project standalone, and it's best if the project can compile by itself (for template build tests, for instance).

The other approach needs modifying the template converter to template `<!-- #if` to be a proper #if, with `server.cs` not having `//-:cnd:noEmit` at the top so it can be interpreted by `dotnet new` properly.

Reply:
@jeromelaban To make sure I understand, is there something I need to do?

Reply:
It would be best not to use the `DefineConstants` and use an actual `dotnet new` variable to conditionally include the code.

Reply:
@jeromelaban just looking at the template.json and the behavior doesn't match for the Server project... you have the `hosted` symbol, but in the modifiers, the Server is excluded when `WebAssembly` is false. There is no modifier for when `hosted` is false.

![image](https://user-images.githubusercontent.com/3860573/217010833-47d611cd-a606-44a0-9d5c-901220f3a416.png)